physical/azure/azure.go

```diff
@@ -167,22 +178,24 @@ func (a *AzureBackend) Get(ctx context.Context, key string) (*physical.Entry, er
 	a.permitPool.Acquire()
 	defer a.permitPool.Release()
 
-	blob := &storage.Blob{
-		Container: a.container,
-		Name:      key,
-	}
-	exists, err := blob.Exists()
+	blobURL := a.container.NewBlockBlobURL(key)
+	res, err := blobURL.Download(ctx, 0, 0, azblob.BlobAccessConditions{}, false)
 	if err != nil {
+		var e azblob.StorageError
+		if errors.As(err, &e) {
+			switch e.ServiceCode() {
+			case azblob.ServiceCodeBlobNotFound:
+				return nil, nil
+			default:
+				return nil, errwrap.Wrapf(fmt.Sprintf("failed to download blob %q: {{err}}", key), err)
+			}
+		}
 		return nil, err
 	}
-	if !exists {
-		return nil, nil
-	}
 
-	reader, err := blob.Get(nil)
-	if err != nil {
-		return nil, err
-	}
+	// TODO how many retries?
+	reader := res.Body(azblob.RetryReaderOptions{MaxRetryRequests: 3})
```

Comment:
Was the old behavior retrying at all?

Reply:
No, it wasn't, see [here](https://github.com/Azure/azure-sdk-for-go/blob/master/storage/blob.go#L585)

Reply:
It should be fine to have no retry and signal failure if it fails to perform the operation the first time around, and this looks like the expectation on storage backend operations behavior in general. It feels a bit non-consistent to retry on Get but not for other operations.

Reply:
Agreed, I removed the retries